framework/bin/gen_tests.pl

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/env perl
+#
+#-------------------------------------------------------------------------------
+# Copyright (c) 2014-2018 René Just, Darioush Jalali, and Defects4J contributors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#-------------------------------------------------------------------------------
+
+=pod
+
+=head1 NAME
+
+gen_tests.pl -- generate a test suite using one of the supported test generators.
+
+=head1 SYNOPSIS
+
+  gen_tests.pl -g generator -p project_id -v version_id -n test_id -o out_dir -b budget [-t tmp_dir] [-D]
+
+=head1 OPTIONS
+
+=over 4
+
+=item -g C<generator>
+
+The test generator to use. Run the following command to see a list of supported
+test generators: C<run_test_gen.pl -g help>
+
+=item -p C<project_id>
+
+Generate tests for this project id.
+See L<Project|Project/"Available Project IDs"> module for available project IDs.
+
+=item -v C<version_id>
+
+Generate tests for this version id.
+Format: C<\d+[bf]>.
+
+=item -n C<test_id>
+
+The id of the generated test suite (i.e., which run of the same configuration).
+
+=item -o F<out_dir>
+
+The root output directory for the generated test suite. The test suite and logs are
+written to:
+F<out_dir/project_id/version_id>.
+
+=item -b C<budget>
+
+The time in seconds allowed for test generation.
+
+=item -t F<tmp_dir>
+
+The temporary root directory to be used to check out the program version (optional).
+The default is F</tmp>.
+
+=item -D
+
+Debug: Enable verbose logging and do not delete the temporary check-out directory
+(optional).
+
+=back
+
+=head1 DESCRIPTION
+
+This script runs the specified test generator on a particular program version.
+Tests are generated for all classes touched by at least one triggering test.
```

Comment:
People might want to generate test cases for different sets of classes. Could you please make this an optional parameter? I would say that the script should by default generate test cases for all modified classes but it should also allow people to provide a custom list of classes.

Reply:
Agreed